### PR TITLE
8281569: Create tests for Frame.setMinimumSize() method

### PR DESCRIPTION
**Pull request by srmandal**
Create tests for setMinimumSize functionality
Create a test which verifies that on increasing the Frame's minimumSize, it gets reflected in the subsequent getSize call
Create a test which verifies that on frame resizes back to minimumSize on calling pack.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8281569](https://bugs.openjdk.java.net/browse/JDK-8281569): Create tests for Frame.setMinimumSize() method

### Reviewers
 * [Alexey Ivanov](https://openjdk.java.net/census#aivanov) (@aivanov-jdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7444/head:pull/7444` \
`$ git checkout pull/7444`

Update a local copy of the PR: \
`$ git checkout pull/7444` \
`$ git pull https://git.openjdk.java.net/jdk pull/7444/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7444`

View PR using the GUI difftool: \
`$ git pr show -t 7444`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7444.diff">https://git.openjdk.java.net/jdk/pull/7444.diff</a>

</details>

### CODE REVIEW

**Comment by srmandal** [comment]
Can someone review this PR?

File: test/jdk/java/awt/Frame/SetMinimumSizeTest/SetMinimumSizeTest1.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import java.awt.Robot;
+
+public class SetMinimumSizeTest1 {
+
+    private static Frame frame;
+    private static Dimension dimension;
+
+    public static void createGUI(Frame _frame) {
+        frame.add(new Button("Button"));
+        frame.setSize(140,140);
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/SetMinimumSizeTest/SetMinimumSizeTest1.java:44]:
```suggestion
        frame.setSize(140, 140);
```

File: test/jdk/java/awt/Frame/SetMinimumSizeTest/SetMinimumSizeTest1.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            }
+        }
+        finally {
+            EventQueue.invokeAndWait(() -> frame.dispose());
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+
+        SetMinimumSizeTest1.doTest();
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/SetMinimumSizeTest/SetMinimumSizeTest1.java:83]:
```suggestion
        doTest();
```

File: test/jdk/java/awt/Frame/SetMinimumSizeTest/SetMinimumSizeTest2.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        finally {
+            EventQueue.invokeAndWait(() -> frame.dispose());
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+
+        SetMinimumSizeTest2.doTest();
+
+    }
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/SetMinimumSizeTest/SetMinimumSizeTest2.java:91]:
```suggestion
    public static void main(String[] args) throws Exception {
        doTest();
    }
```

File: test/jdk/java/awt/Frame/SetMinimumSizeTest/SetMinimumSizeTest1.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+import java.awt.Button;
+import java.awt.Dimension;
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Robot;
+
+public class SetMinimumSizeTest1 {
+
+    private static Frame frame;
+    private static Dimension dimension;
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/SetMinimumSizeTest/SetMinimumSizeTest1.java:40]: The `dimension` field should be volatile as it's accessed from different threads.

File: test/jdk/java/awt/Frame/SetMinimumSizeTest/SetMinimumSizeTest1.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+                dimension.width = dimension.width + 20;
+                dimension.height = dimension.height + 20;
+                frame.setMinimumSize(dimension);
+                frame.invalidate();
+                frame.validate();
+            });
+
+            robot.waitForIdle();
+
+            Dimension actualDimension = frame.getSize();
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/SetMinimumSizeTest/SetMinimumSizeTest1.java:68]: Let's be consistent: this should also be called on EDT.

File: test/jdk/java/awt/Frame/SetMinimumSizeTest/SetMinimumSizeTest1.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        frame.setVisible(true);
+    }
+
+    public static void doTest() throws Exception {
+        try {
+            frame = new Frame();
+            EventQueue.invokeAndWait(() -> createGUI(frame));
+
+            Robot robot = new Robot();
+            robot.setAutoDelay(100);
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/SetMinimumSizeTest/SetMinimumSizeTest1.java:55]: And add
```
robot.waitForIdle();
```

File: test/jdk/java/awt/Frame/SetMinimumSizeTest/SetMinimumSizeTest2.java [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+        frame.setVisible(true);
+    }
+
+    public static void doTest() throws Exception {
+        try {
+            frame = new Frame();
+            EventQueue.invokeAndWait(() -> createGUI(frame));
+
+            Robot robot = new Robot();
+            robot.setAutoDelay(100);
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/SetMinimumSizeTest/SetMinimumSizeTest2.java:55]: Also add
```
robot.waitForIdle();
```

File: test/jdk/java/awt/Frame/SetMinimumSizeTest/SetMinimumSizeTest2.java [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+            EventQueue.invokeAndWait(() -> {
+                frame.pack();
+                frame.invalidate();
+                frame.validate();
+            });
+
+            robot.waitForIdle();
+
+            Dimension actualDimension = frame.getSize();
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/SetMinimumSizeTest/SetMinimumSizeTest2.java:74]: Let's be consistent and access `frame` on EDT only.

**Review by aivanov-jdk — CHANGES_REQUESTED** [review]

File: test/jdk/java/awt/Frame/SetMinimumSizeTest/SetMinimumSizeTest1.java [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+import java.awt.Button;
+import java.awt.Dimension;
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Robot;
+
+public class SetMinimumSizeTest1 {
+
+    private static Frame frame;
+    private static volatile Dimension dimension, actualDimension;
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/SetMinimumSizeTest/SetMinimumSizeTest1.java:40]: In Java, each field should be declared on its own line.

File: test/jdk/java/awt/Frame/SetMinimumSizeTest/SetMinimumSizeTest1.java [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+
+    public static void createGUI(Frame _frame) {
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/SetMinimumSizeTest/SetMinimumSizeTest1.java:42]: The `_frame` parameter is never used, please remove it.

File: test/jdk/java/awt/Frame/SetMinimumSizeTest/SetMinimumSizeTest1.java [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+            EventQueue.invokeAndWait(() -> createGUI(frame));
+
+            Robot robot = new Robot();
+            robot.setAutoDelay(100);
+            robot.waitForIdle();
+
+            EventQueue.invokeAndWait(() -> {
+                dimension = frame.getSize();
+                dimension.width = dimension.width + 20;
+                dimension.height = dimension.height + 20;
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/SetMinimumSizeTest/SetMinimumSizeTest1.java:61]:
```suggestion
                dimension.width += 20;
                dimension.height += 20;
```

File: test/jdk/java/awt/Frame/SetMinimumSizeTest/SetMinimumSizeTest1.java [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+
+            robot.waitForIdle();
+
+            if (!actualDimension.equals(dimension)) {
+                throw new RuntimeException("Test Failed\n"
+                    + "expected dimension:(" + dimension.width + "," + dimension.height +")\n"
+                    + "actual dimension:(" + actualDimension.width + "," + actualDimension.height + ")");
+            }
+        }
+        finally {
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/SetMinimumSizeTest/SetMinimumSizeTest1.java:81]:
```suggestion
        } finally {
```

The finally-block as well as catch- and else-blocks should be at the same line as the closing brace from the previous block.

**Review by aivanov-jdk — CHANGES_REQUESTED** [review]

**Review by aivanov-jdk — APPROVED** [review]

**Comment by srmandal** [comment]
@aivanov-jdk Can you please sponsor my changes?.

**Pull request closed without merging**